Xamarin.PropertyEditing.Mac/Controls/DateExtensions.cs

```diff
@@ -32,8 +32,12 @@ public static NSDate ToNSDate (this DateTime dt)
 		/// <returns>A DateTime</returns>
 		public static DateTime ToDateTime (this NSDate nsDate)
 		{
-			// We loose granularity below millisecond range but that is probably ok
-			return _nsRef.AddSeconds (nsDate.SecondsSinceReferenceDate);
+			try {
+				// We loose granularity below millisecond range but that is probably ok
+				return _nsRef.AddSeconds (nsDate.SecondsSinceReferenceDate);
+			} catch (Exception) {
+				return _nsRef;
```

Comment:
Because in lower dates than minimum date it throws an Exception (I checked this)

Reply:
What's the "minimum" in this case? Can we catch the _specific_ exception it throws instead of a blanket catch?

Reply:
@netonjm can we not check that the entered date isn't below minimum (and assume we need a similar check for above maximum) to avoid the exception and just return the min/max with a ding sound?

Reply:
@netonjm from some quick testing this morning it looks like an `ArgumentOutOfRangeException` is thrown, which makes sense as I tried to enter a negative value and sounds like it would be the most likely exception to be thrown in a Datetime field.
See screenshot below..
<img width="665" alt="Screen Shot 2019-03-12 at 11 14 52" src="https://user-images.githubusercontent.com/271363/54196095-55765380-44b8-11e9-9570-123410313936.png">
Is there another scenario where it is some other exception? If so, what steps are required to trigger this different scenario?